docs/userguide/reporters.md

```diff
@@ -161,5 +161,18 @@ Some of the information in debug log might be redundant.
 > - db object names
 > - etc.
 
- 
-       
\ No newline at end of file
+# Custom reporters
+
+It is possible to add your own reporters by creating an appropriate object type. In principle, it has to be a subtype of `ut_reporter_base`. However, in most cases, it is more convenient to create the custom reporter type under a more specialized type, like `ut_documentation_reporter` or `ut_junit_reporter`, and override just some of the functionality.
```

Comment:
We need to distinguish between `ut_reporter_base` which is a foundation for any reporter and 
`ut_output_reporter_base` which should be used as a base for reporter that is expected to produce output consumable by client (screen/file) outside of database.

Currently, all reporters in utPLSQL framework provide outputs to customer

If however you would like to create a reporter that saves data to a DB table for example, you should use `ut_reporter_base` 

utPLSQL will behave differently based on what type they are build on.

For reporters built on `ut_output_reporter_base` utPLSQL will consume output of the reporter and put it to screen before finishing the `ut.run()` call. For reporters not based on that type, output will not be consumed.

Reply:
You could build a reporter based on `ut_console_reporter_base` if you would like to produce a text output from reporter in different colors

Reply:
Good points, thanks for the review. I'll improve the paragraph as suggested. I hope to do it over the weekend, but can't promise anything. I'm also going to mention where the core reporter types can be found in the repo.

Reply:
Thanks Michał
I rally know how difficult it can be to find time & space to spend on writing some code outside of your da-to-day work. We appreciate all, even smallest contributions. They really make a difference.

Reply:
@jgebal I force-pushed an updated version.